File: backend/src/main/java/com/back/ourlog/domain/content/dto/ContentSearchResultDto.java
package com.back.ourlog.domain.content.dto;

import com.back.ourlog.domain.content.entity.ContentType;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import lombok.Builder;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;

@JsonTypeInfo(
        use = JsonTypeInfo.Id.CLASS,
        include = JsonTypeInfo.As.PROPERTY,
        property = "@class"
)
@Builder
public record ContentSearchResultDto(
        String externalId,
        String title,
        String creatorName,
        String description,
        String posterUrl,
        LocalDateTime releasedAt,
        ContentType type,
        List<String> genres
) implements Serializable {

    @JsonCreator
    public ContentSearchResultDto(
            @JsonProperty("externalId") String externalId,
            @JsonProperty("title") String title,
            @JsonProperty("creatorName") String creatorName,
            @JsonProperty("description") String description,
            @JsonProperty("posterUrl") String posterUrl,
            @JsonProperty("releasedAt") LocalDateTime releasedAt,
            @JsonProperty("type") ContentType type,
            @JsonProperty("genres") List<String> genres
    ) {
        this.externalId = externalId;
        this.title = title;
        this.creatorName = creatorName;
        this.description = description;
        this.posterUrl = posterUrl;
        this.releasedAt = releasedAt;
        this.type = type;
        this.genres = genres;
    }
}

File: backend/src/main/java/com/back/ourlog/domain/content/dto/ContentSearchResultDto.kt
package com.back.ourlog.domain.content.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.back.ourlog.domain.content.entity.ContentType
import java.time.LocalDateTime

@JsonIgnoreProperties(ignoreUnknown = true)
data class ContentSearchResultDto(
    @JsonProperty("externalId")
    val externalId: String,

    @JsonProperty("title")
    val title: String? = null,

    @JsonProperty("creatorName")
    val creatorName: String? = null,

    @JsonProperty("description")
    val description: String? = null,

    @JsonProperty("posterUrl")
    val posterUrl: String? = null,

    @JsonProperty("releasedAt")
    val releasedAt: LocalDateTime? = null,

    @JsonProperty("type")
    val type: ContentType,

    @JsonProperty("genres")
    val genres: List<String> = emptyList()
)